lib/std/Build/Step/Compile.zig

```diff
@@ -1480,6 +1481,19 @@ fn make(step: *Step, prog_node: *std.Progress.Node) !void {
     var prev_search_strategy: SystemLib.SearchStrategy = .paths_first;
     var prev_preferred_link_mode: std.builtin.LinkMode = .Dynamic;
 
+    const compdb_entries_dir: ?[]const u8 = if (b.enable_compdb)
+        b.makeTempPath()
+    else
+        null;
+
+    var compdb_entry_paths_set: ?StringHashMap(void) = if (b.enable_compdb)
+        StringHashMap(void).init(b.allocator)
+    else
+        null;
+    if (compdb_entry_paths_set) |set| {
+        errdefer set.deinit();
+    }
```

Comment:
fyi, this is a no-op because `errdefer` is block-scoped. You'd have to invert the nesting. E.g.
```suggestion
    errdefer {
        if (compdb_entry_paths_set) |set|
            set.deinit();
    }
```

Reply:
@Inve1951 That was an innocent mistake due to tiredness, but thank you.